pkg/cmd/server/origin/authorizer.go

```diff
@@ -51,23 +51,27 @@ func newAuthorizer(
 	podInformer coreinformers.PodInformer,
 	pvInformer coreinformers.PersistentVolumeInformer,
 	projectRequestDenyMessage string,
+
 ) (kauthorizer.Authorizer, authorizer.SubjectLocator) {
 	messageMaker := authorizer.NewForbiddenMessageResolver(projectRequestDenyMessage)
-	roleBasedAuthorizer := authorizer.NewAuthorizer(kubeAuthorizer, messageMaker)
 	subjectLocator := authorizer.NewSubjectLocator(kubeSubjectLocator)
-
-	scopeLimitedAuthorizer := scope.NewAuthorizer(roleBasedAuthorizer, clusterRoleGetter, messageMaker)
-	// Wrap with an authorizer that detects unsafe requests and modifies verbs/resources appropriately so policy can address them separately
-	browserSafeAuthorizer := browsersafe.NewBrowserSafeAuthorizer(scopeLimitedAuthorizer, user.AllAuthenticated)
+	scopeLimitedAuthorizer := scope.NewAuthorizer(clusterRoleGetter, messageMaker)
 
 	graph := node.NewGraph()
 	node.AddGraphEventHandlers(graph, podInformer, pvInformer)
 	nodeAuthorizer := node.NewAuthorizer(graph, nodeidentifier.NewDefaultNodeIdentifier(), kbootstrappolicy.NodeRules())
 
 	authorizer := authorizerunion.New(
-		authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup), // authorizes system:masters to do anything, just like upstream
+		// Wrap with an authorizer that detects unsafe requests and modifies verbs/resources appropriately so policy can address them separately.
+		// Scopes are first because they will authoritatively deny and can logically be attached to anyone.
+		browsersafe.NewBrowserSafeAuthorizer(scopeLimitedAuthorizer, user.AllAuthenticated),
+
+		// authorizes system:masters to do anything, just like upstream
+		authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup),
+		// authorizes only nodes
 		nodeAuthorizer,
-		browserSafeAuthorizer,
```

Comment:
same q for node authorizer

Reply:
> same q for node authorizer

I think that scopes should be respected no matter which user they are provided for.